index.js

```diff
@@ -57,6 +57,18 @@ FileBin.prototype.write = function (fileName, data) {
   });
 };
 
+FileBin.prototype.rename = function (oldFileName, newFileName) {
+  var oldFullPath = path.join(this.base, oldFileName);
+  var newFullPath = path.join(this.base, newFileName);
+  return new RSVP.Promise((resolve, reject) => {
+    fs.rename(oldFullPath, newFullPath, (error) => {
+      if (error) { reject(error); }
+      let file = this.find(newFileName);
+      resolve(file, newFullPath, oldFullPath);
```

Comment:
@stevekinney is this what you meant by returning the file object as the first argument?